### PR TITLE
Remove username/password from elastic-agent

### PR DESCRIPTION
**Pull request by michel-laterman**
Remove username/password env vars for the elastic-agent as it is removed from use in fleet-server.
PRs:
- https://github.com/elastic/fleet-server/pull/1006
- https://github.com/elastic/beats/pull/29458

deprecation PR: https://github.com/elastic/observability-docs/pull/1409

Issue: https://github.com/elastic/fleet-server/issues/978
Closes: https://github.com/elastic/fleet-server/issues/978

### CODE REVIEW

File: docs/en/ingest-management/elastic-agent/configuration/env/shared-env.asciidoc [hunk 1]
```diff
@@ -407,22 +376,6 @@ Setting this to `true` is not recommended.
...
-
-// To do: link to required privileges
-
-*Default:* `elastic`
-
-// end::elasticsearch-username[]
-
-// =============================================================================
-
 // tag::es-username[]
```
Comment (michel-laterman) [inline docs/en/ingest-management/elastic-agent/configuration/env/shared-env.asciidoc:379]: The `es-username` tag and `es-password` tag are not being removed as they are used by the standalone kubernetes config (i think for the agent to send data into es)

File: docs/en/ingest-management/elastic-agent/configuration/env/shared-env.asciidoc [hunk 2]
```diff
@@ -54,8 +54,8 @@ Overrides `FLEET_HOST` when set.
 [id="env-{type}-kibana-fleet-username"]
 `KIBANA_FLEET_USERNAME`
 
-| (string) The basic authentication username used to connect to {kib} and enable {fleet}.
-Overrides `KIBANA_USERNAME` when set.
+| (string) The basic authentication username used to connect to {kib} and retrieve a `service_token` to enable {fleet}.
```
Comment (michel-laterman) [inline docs/en/ingest-management/elastic-agent/configuration/env/shared-env.asciidoc:57]: @dedemorton, I've changed the behaviour of `KIBANA_FLEET_USERNAME` and `ELASTICSEARCH_USERNAME` with https://github.com/elastic/beats/pull/29651, should I update the deprecation tags I added for 7.17 to reflect this?

Comment (dedemorton) [reply docs/en/ingest-management/elastic-agent/configuration/env/shared-env.asciidoc:57]: Yes, if you don't mind. Thanks!

**Comment by jlind23** [comment]
@dedemorton @michel-laterman are we good to merge this PR then?

**Comment by dedemorton** [comment]
No, just give me a little time today to do a proper review. Thanks!

**Review by dedemorton — APPROVED** [review]
LGTM! Thanks for updating the docs.

**Pull request merged**